shodan/__main__.py

```diff
@@ -799,6 +801,87 @@ def _create_stream(name, args, timeout):
+@click.option('--separator', help='The separator between the properties of the search results.', default='\t')
+@click.option('--facets', help='List of facets to get summary information on.', required=True, type=str)
+@click.argument('query', metavar='<search query>', nargs=-1)
+def trends(filename, save, separator, facets, query):
+    """Search Shodan historical database"""
+    key = get_api_key()
+    api = shodan.Shodan(key)
+
+    # Create the query string out of the provided tuple
+    query = ' '.join(query).strip()
+    facets = facets.strip()
+
+    # Make sure the user didn't supply an empty query or facets
+    if query == '':
+        raise click.ClickException('Empty search query')
+
+    if facets == '':
+        raise click.ClickException('Empty search facets')
+
+    # Convert comma-separated facets string to list
+    parsed_facets = []
+    for facet in facets.split(','):
+        parts = facet.strip().split(":")
+        if len(parts) > 1:
+            parsed_facets.append((parts[0], parts[1]))
+        else:
+            parsed_facets.append((parts[0]))
+
+    # Perform the search
+    try:
+        results = api.trends.search(query, facets=parsed_facets)
+    except shodan.APIError as e:
+        raise click.ClickException(e.value)
+
+    # Error out if no results were found
+    if results['total'] == 0:
+        raise click.ClickException('No search results found')
+
+    result_facets = list(results['facets'].keys())
+
+    # Save the results first to file if user request
+    if filename or save:
+        if not filename:
+            filename = '{}-trends.json.gz'.format(query.replace(' ', '-'))
+        elif not filename.endswith('.json.gz'):
+            filename += '.json.gz'
+
+        # Create/ append to the file
+        with helpers.open_file(filename) as fout:
+            for index, match in enumerate(results['matches']):
+                # Append facet info to make up a line
+                match["facets"] = {}
+                for facet in result_facets:
+                    match["facets"][facet] = results['facets'][facet][index]['values']
+                    line = json.dumps(match) + '\n'
+                    fout.write(line.encode('utf-8'))
+
+        click.echo(click.style(u'Saved results into file {}'.format(filename), 'green'))
+
+    # We buffer the entire output so we can use click's pager functionality
+    output = u''
+
+    # Output example:
+    # 2017-06
+    #   os
+    #     Linux 3.x   384148
+    #     Windows 7 or 8      25531
+    for index, match in enumerate(results['matches']):
+        output += click.style(match['month'] + u'\n', fg='green')
+
+        for facet in result_facets:
+            output += '  ' + facet + u'\n'
+            for bucket in results['facets'][facet][index]['values']:
+                output += '    ' + str(bucket['value']) + separator + str(bucket['count']) + u'\n'
```

Comment:
1. It would be better to use fixed-length strings in the format (ex. `'{:20} {:40}'.format(...)`) to make sure that the output is aligned. In your example it's a bit awkward how the count for the `Linux 3.x` facet overlaps with the `Windows 7 or 8` string below.
2. We could make the values bold to separate the value from the count. Ex. ``**Linux 3.x**     384,148``
3. Use `{:xx,d}` in the formatting of the ``count`` so it's easier to read with commas in between every 3rd digit (i.e. ``100,000`` instead of ``100000``)

Reply:
New output format

![Screenshot 2023-07-03 at 10 59 00](https://github.com/achillean/shodan-python/assets/15308084/f7372033-4203-43e8-ba66-9f6733f2270c)